tests/misc-test.cc

```diff
@@ -11,6 +11,15 @@
 
 using namespace CHERI;
 
+/*
+ * Partially ensure that CHERI::Capability<>s can be housed in registers rather
+ * than needing to go via memory.  This is an attempt to capture some of C++'s
+ * [class.temporary]p3 requirements, but I don't know how to capture all of
+ * them in the existing metaprogramming library.
+ */
+static_assert(std::is_trivially_copy_constructible_v<Capability<void>> &&
+              std::is_trivially_destructible_v<Capability<void>>);
+
```

Comment:
Is there a reason to not put this in cheri.hh, where it would fire as soon as you (okay, I) break it?